presto-main/src/main/java/com/facebook/presto/operator/BigintGroupByHash.java

```diff
@@ -267,6 +272,13 @@ private void rehash()
         }
         int newCapacity = (int) newCapacityLong;
 
+        // An estimation of how much extra memory needed before allocating the hash table
+        if (!updateReservedMemory.apply(((long) (newCapacity - hashCapacity)) * (Long.BYTES + Integer.BYTES))) {
```

Comment:
I guess _during_ rehashing you need both old capacity and new capacity.

Reply:
I thought about that; but that may possibly fail existing queries in production that are running very close to memory limit. So maybe be conservative for now?